tests/input/inline_type_hint_assertions.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+final class Foo
+{
+    public function methodWithInlineHints()
+    {
+        /** @var Type $simpleType */
+        $simpleType = expression();
+
+        $typeDeclaredAfterExpression = expression();
+        /** @var Type $typeDeclaredAfterExpression */
+
+        $typeDeclaredViaAssertion = expression();
+        assert($typeDeclaredViaAssertion instanceof Type);
+
+        /** @var Type1|Type2 $unionType */
+        $unionType = expression();
+
+        /** @var Type1&Type2 $intersectionType */
+        $intersectionType = expression();
+
+        /** @var Type|null $nullableType */
+        $nullableType = expression();
+
+        /** @var int|float|bool|string|null|array $multipleScalarTypes */
+        $multipleScalarTypes = expression();
+
+        /** @var Potato $variableThatIsNowhereToBeFound */
```

Comment:
Is it for the cases like `extract()` or `eval()`? If the standard doesn't allow this case (and why would it?), then this case doesn't need to be handled.

Reply:
Replacing an inline definition with an assertion still challenges the assumption, which is good